packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts

```diff
@@ -211,7 +257,33 @@ export class BedrockInvokeModel extends sfn.TaskStateBase {
         Output: this.props.output?.s3Location ? {
           S3Uri: `s3://${this.props.output.s3Location.bucketName}/${this.props.output.s3Location.objectKey}`,
         } : undefined,
+        GuardrailIdentifier: this.props.guardrailConfiguration?.guardrailIdentifier,
+        GuardrailVersion: this.props.guardrailConfiguration?.guardrailVersion,
+        Trace: this.props.traceEnabled === undefined
+          ? undefined
+          : this.props.traceEnabled
+            ? 'ENABLED'
+            : 'DISABLED',
       }),
     };
   }
+
+  private validateGuardrailConfiguration(props: BedrockInvokeModelProps) {
+    if (props.guardrailConfiguration === undefined
+      || Token.isUnresolved(props.guardrailConfiguration.guardrailIdentifier)
+      || Token.isUnresolved(props.guardrailConfiguration.guardrailVersion)) { return; }
+
+    if (!/^(([a-z0-9]+)|(arn:aws(-[^:]+)?:bedrock:[a-z0-9-]{1,20}:[0-9]{12}:guardrail\/[a-z0-9]+))$/.test(props.guardrailConfiguration.guardrailIdentifier)) {
+      throw new Error(`guardrailIdentifier must match the ^(([a-z0-9]+)|(arn:aws(-[^:]+)?:bedrock:[a-z0-9-]{1,20}:[0-9]{12}:guardrail/[a-z0-9]+))$ pattern, got ${props.guardrailConfiguration.guardrailIdentifier}`);
+    }
+
+    if (props.guardrailConfiguration.guardrailVersion && !/^(([1-9][0-9]{0,7})|(DRAFT))$/.test(props.guardrailConfiguration.guardrailVersion)) {
+      throw new Error(`guardrailVersion must match the ^(([1-9][0-9]{0,7})|(DRAFT))$ pattern, got ${props.guardrailConfiguration.guardrailVersion}`);
+    }
+
+    if (props.guardrailConfiguration.guardrailIdentifier && (props.contentType !== 'application/json')) {
+      throw new Error('You must set contentType to \'application/json\' when use guardrailConfiguration');
+    }
```

Comment:
```suggestion
  private validateGuardrailConfiguration(props: BedrockInvokeModelProps) {
    if (!props.guardrailConfiguration) return;

    const { guardrailIdentifier, guardrailVersion } = props.guardrailConfiguration;

    if (!Token.isUnresolved(guardrailIdentifier)) {
      const guardrailConfigurationPattern = /^(([a-z0-9]+)|(arn:aws(-[^:]+)?:bedrock:[a-z0-9-]{1,20}:[0-9]{12}:guardrail\/[a-z0-9]+))$/;
      if (!guardrailConfigurationPattern.test(guardrailIdentifier)) {
        throw new Error(`guardrailIdentifier must match the \`${guardrailConfigurationPattern}\` pattern, got ${guardrailIdentifier}`);
      }
      if (props.contentType !== 'application/json') {
        throw new Error(`You must set contentType to \'application/json\' when using guardrailConfiguration, got ${props.contentType}.`);
      }
    }

    const guardrailVersionPattern = /^(([1-9][0-9]{0,7})|(DRAFT))$/;
    if (!Token.isUnresolved(guardrailVersion) && !guardrailVersionPattern.test(guardrailVersion)) {
      throw new Error(`guardrailVersion must match the \`${guardrailVersionPattern}\` pattern, got ${guardrailVersion}`);
    }
  }
```
We should validate the two props independently

Reply:
Thank you. 
The method you suggested includes `/` around the regular expression in the error message, so I wrote it directly in the error message instead. 
If you have a better way, please let me know.